Snippets/Extensions.Hosting/Extensions.Hosting_1/UnsubscribeService.cs

```diff
@@ -0,0 +1,31 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using NServiceBus;
+
+#region UnsubscribeBackgroundService
+public sealed class UnsubscribeService : BackgroundService
+{
+    readonly IMessageSession _messageSession;
+
+    public UnsubscribeService(IMessageSession messageSession)
+    {
+        _messageSession = messageSession;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        await _messageSession.Unsubscribe<MyEvent>();
+    }
+}
+
+// var host = Host.CreateDefaultBuilder(args)
+//  .ConfigureServices(services => {
+//    services.AddHostedService<UnsubscribeService>();
+//  })
+//  .Build();
```

Comment:
This is part of the region, indicating that it will be shown in the snippet, should this be moved out or remove the commented code?

Reply:
That is intentional, to show how to register it but maybe we can assume this should be known. I'll remove it.